kafka/src/main/java/com/linecorp/armeria/common/logging/kafka/KafkaLogWriter.java

```diff
@@ -0,0 +1,161 @@
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.logging.kafka;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Function;
+
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.common.logging.LogFormatter;
+import com.linecorp.armeria.common.logging.LogWriter;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.common.logging.RequestOnlyLog;
+
+/**
+ * A {@link LogWriter} that sends request and response logs to a Kafka backend.
+ *
+ * <p>This implementation uses {@link LogFormatter#ofJson()} by default to serialize logs
+ * as JSON before sending to Kafka. You can customize the serialization by providing
+ * custom key and value extractors.</p>
+ *
+ * <p>Example usage:</p>
+ * <pre>{@code
+ * Producer<String, String> producer = new KafkaProducer<>(props);
+ * LogWriter logWriter = new KafkaLogWriter<>(producer, "armeria-logs");
+ * 
+ * Server server = Server.builder()
+ *     .service("/", (ctx, req) -> HttpResponse.of("Hello!"))
+ *     .decorator(LoggingService.builder()
+ *         .logWriter(logWriter)
+ *         .newDecorator())
+ *     .build();
+ * }</pre>
+ *
+ * @param <K> the type of the Kafka record key
+ * @param <V> the type of the Kafka record value
+ */
+@UnstableApi
+public final class KafkaLogWriter<K, V> implements LogWriter {
+
+    private static final Logger logger = LoggerFactory.getLogger(KafkaLogWriter.class);
+
+    private final Producer<K, V> producer;
+    private final String topic;
+    private final Function<? super RequestOnlyLog, ? extends @Nullable K> requestKeyExtractor;
+    private final Function<? super RequestOnlyLog, ? extends @Nullable V> requestValueExtractor;
+    private final Function<? super RequestLog, ? extends @Nullable K> responseKeyExtractor;
+    private final Function<? super RequestLog, ? extends @Nullable V> responseValueExtractor;
+
+    /**
+     * Creates a new instance with JSON log formatting.
+     *
+     * @param producer a Kafka {@link Producer} which is used to send logs to Kafka
+     * @param topic the name of topic which is used to send logs
+     */
+    @SuppressWarnings("unchecked")
+    public KafkaLogWriter(Producer<K, V> producer, String topic) {
```

Comment:
Since there is a `*Builder` class, I'm guessing the constructors don't need to be `public`

```suggestion
    KafkaLogWriter(Producer<K, V> producer, String topic) {
```